File: backend/app/schemas/notification.py
"""Notification schemas."""
from datetime import datetime
from typing import Optional, List
from uuid import UUID

from pydantic import BaseModel


class NotificationResponse(BaseModel):
    """Notification data in responses."""
    id: UUID
    type: str
    title: str
    body: Optional[str] = None
    data: Optional[dict] = None
    sighting_id: Optional[UUID] = None
    is_read: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class NotificationListResponse(BaseModel):
    """Paginated notification list."""
    items: List[NotificationResponse]
    total: int
    unread_count: int

File: web_app/src/types/notification.ts
export interface NotificationItem {
  id: string;
  user_id: string;
  title: string;
  body: string;
  sighting_id?: string | null;
  is_read: boolean;
  data?: Record<string, unknown> | null;
  created_at: string;
}

export interface NotificationListResponse {
  items: NotificationItem[];
  total: number;
  unread_count: number;
}
